actors.md

```diff
@@ -2,10 +2,94 @@
 
 Any implementations of the Filecoin actors must be exactly byte for byte compatible with the go-filecoin actor implementations. The pseudocode below tries to capture the important logic, but capturing all the detail would require embedding exactly the code from go-filecoin, so for now, its simply informative pseudocode. The algorithms below are correct, and all implementations much match it (including go-filecoin), but details omitted from here should be looked for in the go-filecoin code.
 
-This spec decsribes a set of actors that operate within the [Filecoin State Machine](state-machine.md). All types are defined in [the basic type encoding spec](data-structures.md#basic-type-encodings).
+This spec describes a set of actors that operate within the [Filecoin State Machine](state-machine.md). All types are defined in [the basic type encoding spec](data-structures.md#basic-type-encodings).
 
+
+- [Init Actor](#init-actor)
 - [Storage Market Actor](#storage-market-actor)
 - [Storage Miner Actor](#storage-miner-actor)
+- [Payment Channel Broker Actor](#payment-channel-broker-actor)
+
+
+## Init Actor
+The init actor is responsible for creating new actors on the filecoin network. This is a built-in actor and cannot be replicated. In the future, this actor will be responsible for loading new code into the system (for user programmable actors).
+
+```go
+type InitActor struct {
+    // Mapping from Address to ID, for lookups.
+		AddressMap map[Address]BigInt
+}
+```
+
+### `Exec(code Cid, params []Param) Address`
+
+>  This method is the core of the `Init Actor`. It handles instantiating new actors and assigning them their IDs.
+
+#### Parameters
+
+| Name     | Type       | Description                                                  |
+| -------- | ---------- | ------------------------------------------------------------ |
+| `code`   | `Cid`      | A pointer to the location at which the code of the actor to create is stored. |
+| `params` | `[] Param` | The parameters passed to the constructor of the actor.       |
+
+
+
+`Param` is the type representing any valid arugment that can be passed to a function.
+
+TODO: Find a better place for this definition.
+
+
+```go
+func Exec(code Cid, params []byte) Address {
+    // Get the actor ID for this actor.
+    actorID = len(self.AddressMap)
```

Comment:
I think relying on the size of this map to keep track of the next ID may end up being error prone, i'm worried that if we allow actors to be deleted, this will result in unexpected behavior

Reply:
You are not wrong, it just felt nicer to use this given the fact that we don't have deletion atm, but I am fine reverting to keeping track of the id manually